### ISSUE TITLE
Schema object type should handle variable set of fields

### ISSUE THREAD

**Issue by kinabalu** [encode/apistar#152]
Given a possible JSON like the following:

```
{
 "id": 'random-uuid-value',
 "name": "Foo",
 "extra_fields": {
    "random_field_1": "random_value_1",
    "random_field_2": "random_value_2"
  }
}
```
We should be able to use the Schema support to define it something like:

```
class JsonBody(schema.Object):
  properties = {
    'id': schema.String(max_length=36),
    'name': schema.String(min_length=3),
    'extra_fields': schema.Object()
  }
```

**Comment by tomchristie** [comment]
Yup. I punted on this deliberately to start with, because GraphQL doesn't support open mappings, and I was hoping that our type system could also nicely correspond with that, but that's me thinking too far ahead in any case. So yes, to be added.

**Comment by frol** [comment]
@kinabalu Could you, please, share how do you force apistar to expect JSON body?

**Comment by kinabalu** [comment]
it was just an example class, that's not what that block of code actually does

**Comment by tomchristie** [comment]
This is resolved in #400. (See also https://discuss.apistar.org/t/a-peek-at-the-upcoming-version/412)

```python
class JsonBody(types.Type):
    id = validators.String(max_length=36)
    name = validators.String(min_length=3)
    extra_fields = validators.Object()
```

Note that `JsonBody` here is a declarative type of object, whereas `extra_fields` uses the more general purpose validator.
